tensorflow/core/kernels/decode_bmp_op.cc

```diff
@@ -56,6 +56,20 @@ class DecodeBmpOp : public OpKernel {
                                         input.size(), " bytes"));
 
     const uint8* img_bytes = reinterpret_cast<const uint8*>(input.data());
+#if (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
+    int32 header_size_ = internal::SubtleMustCopy(
```

Comment:
I think it would be cleaner to isolate the platform dependent code in a little utility to avoid code duplication:
```
inline int32 MaybeByteSwapInt32(int32 x) {
#if (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
   return le32toh(x);
#else 
  return x;
}
```
and pass the values through that unconditionally.

Reply:
@rmlarsen , Thank you for your comments.
I will try this and commit changes.

Reply:
@rmlarsen, could you please have a look?